pkg/live/inventoryrg.go

```diff
@@ -292,15 +292,36 @@ func (icm *InventoryResourceGroup) ApplyWithPrune(dc dynamic.Interface, mapper m
 	}
 
 	// Update the cluster inventory object.
+	// Since the ResourceGroup CRD specifies the status as a sub-resource, this
+	// will not update the status.
 	appliedObj, err := namespacedClient.Update(context.TODO(), invInfo, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
 
-	// Update status.
+	// Update status, if status policy allows it.
```

Comment:
Some unit tests here would be nice to have if it's not too cumbersome

Reply:
Adding unit tests would require faking the `dynamic.Interface` and `meta.RESTMapper`. The fakes I wrote are in kpt-config-sync downstream, which make them hard to use here. If we want to use the fake clients in kpt, we'll have to upstream the fakes into cli-utils first.

So adding unit tests might be more effort than it's worth at this point.